### PR TITLE
19823 and 19843 - added support for amalg correction + preset numbered option for reg amalg

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#19823 and bcgov/entity#19843

*Description of changes:*
- app version = 7.0.44
- added conditional disable for amalgamation correction
- redirect to Edit UI for amalgamation correction
- cleaned up AmalgamationSelection.vue
- pre-set numbered option for regular amalgamation draft
- updated (but did not fix) unit test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-622-si27jgy1.web.app

File: src/components/Dashboard/FilingHistoryList/HeaderActions.vue [hunk 1]
```diff
@@ -132,6 +132,7 @@ export default class HeaderActions extends Mixins(AllowableActionsMixin) {
       !this.isFirm && !this.isBenBcCccUlc &&
       !this.isCoop)
     conditions[6] = () => (EnumUtilities.isTypeRegistration(this.filing) && !this.isFirm)
+    conditions[7] = () => (EnumUtilities.isTypeAmalgamation(this.filing) && !this.isBenBcCccUlc)
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/HeaderActions.vue:135]: This is a safety check to ensure that the correction action for this ledger item is disabled if this business is not a supported entity type.

File: src/components/Dashboard/FilingHistoryList/HeaderActions.vue [hunk 2]
```diff
@@ -142,6 +143,7 @@ export default class HeaderActions extends Mixins(AllowableActionsMixin) {
     // see also TodoList.vue:doResumeFiling()
     switch (filing?.name) {
       case FilingTypes.ALTERATION:
+      case FilingTypes.AMALGAMATION_APPLICATION:
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/HeaderActions.vue:146]: This is to redirect to Edit UI for a correction.

Without this, the default was used (line 175), which routes to a local correction notation.

File: src/views/AmalgamationSelection.vue [hunk 3]
```diff
@@ -203,16 +203,14 @@ export default class AmalgamationSelection extends Vue {
...
       this.setStartingAmalgamationSpinner(true)
       const businessId = await this.createBusinessAA(amalgamationType)
-      const route =
-        amalgamationType === AmalgamationTypes.HORIZONTAL ||
-        amalgamationType === AmalgamationTypes.VERTICAL
-          ? 'amalg-short-information'
-          : 'amalg-reg-information'
+      const route = this.isShortFormAmalgamation(amalgamationType)
+        ? 'amalg-short-information'
+        : 'amalg-reg-information'
```
Comment (severinbeauvais) [inline src/views/AmalgamationSelection.vue:208]: I created isShortFormAmalgamation() to simplify things.

File: src/views/AmalgamationSelection.vue [hunk 4]
```diff
@@ -225,45 +223,38 @@ export default class AmalgamationSelection extends Vue {
...
-    const legalName =
-      correctNameOption === CorrectNameOptions.CORRECT_AML_ADOPT
-        ? this.getLegalName
-        : null
+    const email = this.isShortFormAmalgamation(type) ? this.getBusinessEmail : ''
+    const phone = this.isShortFormAmalgamation(type) ? this.getFullPhoneNumber : ''
+    const legalName = this.isShortFormAmalgamation(type) ? this.getLegalName : ''
+    const correctNameOption = this.isShortFormAmalgamation(type)
+      ? CorrectNameOptions.CORRECT_AML_ADOPT
+      : CorrectNameOptions.CORRECT_AML_NUMBERED
```
Comment (severinbeauvais) [inline src/views/AmalgamationSelection.vue:231]: This is the change for regular amalgamations. (And there is no legal name for the numbered option.)

**Review by PaulGarewal — APPROVED** [review]
LGTM. Great learning opportunity, especially in AmalgamationSelection 👍

**Review by ketaki-deodhar — APPROVED** [review]
Looks good Sev!

**Review by jamespaologarcia — APPROVED** [review]
LGTM

**Pull request merged**